runtime/src/main/java/org/corfudb/runtime/DistributedCheckpointerHelper.java

```diff
@@ -19,27 +21,75 @@ public class DistributedCheckpointerHelper {
 
     private final CorfuStore corfuStore;
 
-    public DistributedCheckpointerHelper(CorfuStore corfuStore) {
+    @Getter
+    private final CompactorMetadataTables compactorMetadataTables;
+
+    public DistributedCheckpointerHelper(CorfuStore corfuStore) throws Exception {
         this.corfuStore = corfuStore;
+
+        // Open all compactor related tables
+        this.compactorMetadataTables = new CompactorMetadataTables(corfuStore);
     }
 
     public static enum UpdateAction {
         PUT,
         DELETE
     }
 
-    public void updateCompactionControlsTable(Table<StringKey, RpcCommon.TokenMsg, Message> checkpointTable,
-                                              StringKey stringKey,
-                                              UpdateAction action) {
+    public void disableCompaction() {
+        updateCompactionControlsTable(
+                CompactorMetadataTables.DISABLE_COMPACTION, UpdateAction.PUT);
+        log.info("Disabled compaction");
+    }
 
-        try (TxnContext txn = corfuStore.txn(CORFU_SYSTEM_NAMESPACE)) {
-            updateCompactionControlsTable(txn, checkpointTable, stringKey, action);
-            txn.commit();
-        } catch (Exception e) {
-            log.warn("Unable to write UpgradeKey to checkpoint table, ", e);
+    public void enableCompaction() {
+        updateCompactionControlsTable(CompactorMetadataTables.DISABLE_COMPACTION, UpdateAction.DELETE);
+        log.info("Disabled compaction");
+    }
+
+    public void freezeCompaction() {
+        updateCompactionControlsTable(CompactorMetadataTables.FREEZE_TOKEN, UpdateAction.PUT);
+    }
+
+    public void unfreezeCompaction() {
+        updateCompactionControlsTable(CompactorMetadataTables.FREEZE_TOKEN, UpdateAction.DELETE);
+    }
+
+    public void instantTrigger(boolean trim) {
+        if (trim) {
+            updateCompactionControlsTable(CompactorMetadataTables.INSTANT_TIGGER_WITH_TRIM, UpdateAction.PUT);
+        } else {
+            updateCompactionControlsTable(CompactorMetadataTables.INSTANT_TIGGER, UpdateAction.PUT);
+        }
+    }
+
+    private void updateCompactionControlsTable(StringKey stringKey, UpdateAction action) {
+
+        Table<StringKey, RpcCommon.TokenMsg, Message> compactionControlsTable =
+                compactorMetadataTables.getCompactionControlsTable();
+        for (int retry = 1; retry <= CompactorMetadataTables.MAX_RETRIES; retry++) {
+            try (TxnContext txn = corfuStore.txn(CORFU_SYSTEM_NAMESPACE)) {
+                updateCompactionControlsTable(txn, compactionControlsTable, stringKey, action);
+                txn.commit();
```

Comment:
Don't you want to break here otherwise it will run full retry times always right?

Reply:
Done.